contracts/staking/contracts/src/staking_pools/MixinCumulativeRewards.sol

```diff
@@ -58,22 +43,71 @@ contract MixinCumulativeRewards is
         return cumulativeReward.denominator != 0;
     }
 
-    /// @dev Sets a cumulative reward for `poolId` at `epoch`.
-    /// This can be used to overwrite an existing value.
-    /// @param poolId Unique Id of pool.
-    /// @param value Value of cumulative reward.
-    function _forceSetCumulativeReward(
+    /// @dev Sets a pool's cumulative delegator rewards for the current epoch,
+    ///      given the rewards earned and stake from the last epoch, which will
+    ///      be summed with the previous cumulative rewards for this pool.
+    ///      If the last cumulative reward epoch is the current epoch, this is a
+    ///      no-op.
+    /// @param poolId The pool ID.
+    /// @param reward The total reward earned by pool delegators from the last epoch.
+    /// @param stake The total delegated stake in the pool in the last epoch.
+    function _addCumulativeReward(
         bytes32 poolId,
-        IStructs.Fraction memory value
+        uint256 reward,
+        uint256 stake
     )
         internal
     {
+        // Fetch the last epoch at which we stored an entry for this pool;
+        // this is the most up-to-date cumulative rewards for this pool.
+        uint256 lastStoredEpoch = _cumulativeRewardsByPoolLastStored[poolId];
         uint256 currentEpoch_ = currentEpoch;
-        _cumulativeRewardsByPool[poolId][currentEpoch_] = value;
-        // Update state to reflect the most recent cumulative reward
+
+        // If we already have a record for this epoch, don't overwrite it.
+        if (lastStoredEpoch == currentEpoch_) {
+            return;
+        }
+
+        IStructs.Fraction memory mostRecentCumulativeReward =
+            _cumulativeRewardsByPool[poolId][lastStoredEpoch];
+
+        // Compute new cumulative reward
+        IStructs.Fraction memory cumulativeReward;
+        if (_isCumulativeRewardSet(mostRecentCumulativeReward)) {
```

Comment:
In what cases is this false? Is this just if cumulative rewards for a pool have never been set?

Reply:
Yeah, because we don't set a CR on pool creation anymore. So if a pool is created, doesn't earn any rewards, then someone delegates, this function will get invoked, but there will be no prior CR. And since `_getCumulativeRewardAtEpoch()` now returns an empty CR if it can't find any, there's no point copying empty CRs to the places it used to look.